### PR TITLE
[#41945] fixed backups controller permissions and actions

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/41945

### CODE REVIEW

File: app/controllers/admin/backups_controller.rb [hunk 1]
```diff
@@ -111,10 +111,10 @@ def token_reset_flash_message(token)
     ]
   end
 
-  def token_reset_failed!(e)
-    Rails.logger.error "Failed to reset user ##{current_user.id}'s Backup key: #{e}"
+  def token_reset_failed!(key)
```
Comment (machisuji) [inline app/controllers/admin/backups_controller.rb]: This is an exception which is expected here, hence `e`. I think it's rather the log message that's wrong in saying "Backup key:". So I'd just remove the word 'key' from the log message and keep the variable name.

If you look further below you can see that `e.message` is used so we definitely have an exception here.

**Review by machisuji — CHANGES_REQUESTED** [review]
Looks good overall. I commend the extra hygiene on the parts you touched too.

Unfortunately you are not done yet, though!
Resetting the backup token doesn't work after the change from `required_admin?` to `authorize_global` anymore.

image


It's not your fault that this isn't caught by the evidently missing specs for that case.
You'll still have to fix the issue, though. :D If you feel like it, adding the missing specs would be greatly appreciated, of course.

But if you are unsure about this we can have a look at it together.

File: app/controllers/admin/backups_controller.rb (hunk 1 above)
Comment (Kharonus) [reply app/controllers/admin/backups_controller.rb]: ah, then I will of course change the name ;)

Comment (cbliard) [reply app/controllers/admin/backups_controller.rb]: fixed in 4f701c8d48

**Comment by cbliard** [comment]
@machisuji it should be good now. Can you have a look again please?

**Review by machisuji — APPROVED** [review]
All good now. Thanks @Kharonus and @cbliard!

File: config/initializers/menus.rb [hunk 2]
```diff
@@ -364,7 +368,7 @@
 
   menu.push :backups,
             { controller: '/admin/backups', action: 'show' },
-            if: Proc.new { OpenProject::Configuration.backup_enabled? && User.current.admin? },
+            if: Proc.new { OpenProject::Configuration.backup_enabled? && User.current.allowed_to_globally?(Backup.permission) },
```
Comment (cbliard) [inline config/initializers/menus.rb:371]: I forgot to mention that doing `Backup.permission` here during initialization will load `Backup` class and lead to the infamous message `DEPRECATION WARNING: Initialization autoloaded the constants [...], Backup, ...`.

We could do it if the menu initialization was not part of the Rails initializers but we're not there yet.

**Pull request merged**